lib/like-pattern-concordance.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const concordance = require('concordance');
+const isPlainObject = require('is-plain-object');
+
+function pickLikePatternPaths(actual, likePattern, visited = new WeakMap()) {
+	if (visited.has(likePattern)) {
+		return visited.get(likePattern);
+	}
+
+	if (isPlainObject(likePattern) && isPlainObject(actual)) {
```

Comment:
What should we do with arrays, and more complex objects like maps and sets?

Reply:
They are left as is to be compared by concordance, anything that's not a plain object is. I don't have a good argument for why it should be this way, I just went with examples from #845.